### PR TITLE
Rename to project_celltype_metadata

### LINKED ISSUE #562: Remove `add-celltypes.nf` workflow
RIP, former cell typing workflow.

### LINKED ISSUE #576: Rename celltype project metadata parameter
Do we want to rename the parameter so it's `project_celltype_metafile` to be less confusing?

_Originally posted by @allyhawkins in https://github.com/AlexsLemonade/scpca-nf/pull/573#discussion_r1396332978_
            
We do indeed! This way, all files and code are the same word order - `project_celltype_meta<data/file>`

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #576
Closes #562

Here, I change all the `celltype_project_meta<data/file>` ->  `project_celltype_meta<data/file>`.

I found the spots as - 

```
scpca-nf ➤ grep -r "celltype_project_meta" .

./config/profile_stub.config:  **celltype_project_metafile** = 'test/stub-celltype-project-metadata.tsv'
./config/profile_ccdl.config:  **celltype_project_metafile** = "s3://ccdl-scpca-data/sample_info/scpca-project-celltype-metadata.tsv"
./add-celltypes.nf:if (!file(params.**celltype_project_metafile**).exists()) {
./add-celltypes.nf:  log.error("The '**celltype_project_metafile**' file '${params.**celltype_project_metafile**}' can not be found.")
./external-instructions.md:2. [Prepare a `**celltype_project_metafile**` TSV](#preparing-the-cell-type-project-metadata-file) to provide reference dataset information for each of `SingleR` and `CellAssign` to the workflow.
./external-instructions.md:   You will need to provide the path/uri to this file as a workflow parameter (`**celltype_project_metafile**`), which you will need to define in your configuration file.
./external-instructions.md:Once you have followed the above steps and added the path/uri to the `**celltype_project_metafile**` to your configuration file, you can use the following command to run the workflow with cell type annotation:
./external-instructions.md:The `**celltype_project_metafile**` file should contain these five columns with the following information:
./external-instructions.md:| [View example `**celltype_project_metafile**.tsv` file](examples/example_project_celltype_metadata.tsv) |
./modules/classify-celltypes.nf:    celltype_ch = Channel.fromPath(params.**celltype_project_metafile**)
```

I fixed everything except for `add-celltypes.nf`, which I went ahead and removed instead to close out #562. Some styling also from hooks made it into the config files, too.

`grep -r "celltype_project" .` now comes up empty for me!

### CODE REVIEW

**Review by allyhawkins — APPROVED** [review]
🚀

File: external-instructions.md [hunk 1]
```diff
@@ -389,7 +389,7 @@ The `celltype_project_metafile` file should contain these five columns with the
 
 We have provided an example cell type project metadata file for reference.
 
-| [View example `celltype_project_metafile.tsv` file](examples/example_project_celltype_metadata.tsv) |
+| [View example `project_celltype_metafile.tsv` file](examples/example_project_celltype_metadata.tsv) |
```
Comment (jashapiro) [inline external-instructions.md:392]: Do we the others have names like this or should it be `metadata.tsv` here to match the file name

File: external-instructions.md [hunk 2]
```diff
@@ -377,7 +377,7 @@ For example, the reference `blood-compartment` includes cell types categorized i
 ### Preparing the cell type project metadata file
 
 All libraries within a given project will use the same reference dataset for each of `SingleR` and `CellAssign`, respectively.
-The `celltype_project_metafile` file should contain these five columns with the following information:
+The `project_celltype_metafile` file should contain these five columns with the following information:
```
Comment (jashapiro) [inline external-instructions.md:380]: Maybe we don't need to use the variable name here at all? 
`The project celltype metadata file` seems clear enough to me, and I don't think we do this elsewhere.

**Review by jashapiro — COMMENTED** [review]
I'm late, but I think we should change wording to remove `project_celltype_metafile` everywhere that we are not referring to the parameter itself, and just use the words "project celltype metadata file" in other cases.

**Pull request merged**